### PR TITLE
feat(profiling): show onboarding page based on prompt check

### PR DESCRIPTION
**Pull request by JonasBa**
Adding yet another loader (hopefully this gets replaced soon once we have a full onboarding guide)

### CODE REVIEW

**Review by eliashussary — CHANGES_REQUESTED** [review]

File: static/app/views/profiling/profilingOnboarding.tsx [hunk 1]
```diff
@@ -8,13 +9,24 @@ import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {t} from 'sentry/locale';
 import {PageContent} from 'sentry/styles/organization';
 import space from 'sentry/styles/space';
-import useOrganization from 'sentry/utils/useOrganization';
+import {Organization} from 'sentry/types/organization';
+import trackAdvancedAnalyticsEvent from 'sentry/utils/analytics/trackAdvancedAnalyticsEvent';
 
-export function ProfilingOnboarding() {
-  const organization = useOrganization();
+interface Props {
```
Comment (eliashussary) [inline static/app/views/profiling/profilingOnboarding.tsx:15]: is this the generic `Props` name we speak of?

Comment (JonasBa) [reply static/app/views/profiling/profilingOnboarding.tsx:16]: That's the current styleguide... I would prefer to prefix with component name, but we should bring that up at the frontend tsc

File: static/app/views/profiling/profilingOnboarding.tsx [hunk 2]
```diff
@@ -95,8 +101,10 @@ export function ProfilingOnboarding() {
               </li>
 
               <Actions>
-                <Button priority="primary">I'm done</Button>
-                <Button>Dismiss</Button>
+                <Button priority="primary" onClick={props.onDoneClick}>
+                  I'm done
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/views/profiling/profilingOnboarding.tsx]: Rebased and it was added.

**Review by eliashussary — APPROVED** [review]
didn't mean to submit as request changes 😄

File: static/app/views/profiling/index.tsx [hunk 3]
```diff
@@ -1,23 +1,91 @@
...
+    promptsCheck(client, {
+      organizationId: organization.id,
+      feature: 'profiling_onboarding',
+    })
+      .then(data => {
+        setRequestState({type: 'resolved', data});
+      })
+      .catch(e => {
+        Sentry.captureException(e);
+        setRequestState({type: 'errored', error: t('Error: Unable to load prompt data')});
```
[the remark this answers is not in the document]
Comment (eliashussary) [reply static/app/views/profiling/index.tsx:57]: i don't mean dropping the exception, i meant dropping the `setState` since we don't assert against `errored`... but i'm missing the fact we're changing it from `loading` which would allow our condition to fall through.

going to get more coffee ☕

**Comment by JonasBa** [comment]
@eliashussary I think more people should request changes more often actually :D imo you either request changes or approve, there is no middle ground 🗡️

**Comment by JonasBa** [comment]
Removed backend label as it no longer contains backend changes

File: static/app/views/profiling/index.tsx [hunk 4]
```diff
@@ -1,23 +1,96 @@
...
+  // Eagerly update state and update check
+  const dismissPrompt = useCallback(
+    (status: 'done' | 'dismissed') => {
+      setRequestState({type: 'resolved', data: {dismissedTime: Date.now()}});
+      trackAdvancedAnalyticsEvent(
+        status === 'done'
+          ? 'profiling_views.onboarding_action.done'
+          : 'profiling_views.onboarding_action.dismiss',
+        {
+          action: status === 'done' ? 'done' : 'dismiss',
```
Comment (Zylphrex) [inline static/app/views/profiling/index.tsx:70]: nit: this ternary isn't necessary
```suggestion
          action: status,
```

File: static/app/utils/analytics/profilingAnalyticsEvents.tsx [hunk 5]
```diff
@@ -2,6 +2,12 @@ export type ProfilingEventParameters = {
   'profiling_views.go_to_flamegraph': {source: string};
   'profiling_views.landing': {};
   'profiling_views.onboarding': {};
+  'profiling_views.onboarding_action.dismiss': {
+    action: 'dismiss';
+  };
+  'profiling_views.onboarding_action.done': {
+    action: 'done';
+  };
```
Comment (Zylphrex) [inline static/app/utils/analytics/profilingAnalyticsEvents.tsx:10]: thoughts on sending this under the same analytic event like `profiling_views.onboarding_action` and just use `action` to see what was clicked instead of 2 separate events?

File: src/sentry/utils/prompts.py [hunk 6]
```diff
@@ -2,15 +2,16 @@
...
     "distributed_tracing": {"required_fields": ["organization_id", "project_id"]},
-    "quick_trace_missing": {"required_fields": ["organization_id", "project_id"]},
     "code_owners": {"required_fields": ["organization_id", "project_id"]},
+    "profiling_onboarding": {"required_fields": ["organization_id"]},
+    "releases": {"required_fields": ["organization_id", "project_id"]},
+    "quick_trace_missing": {"required_fields": ["organization_id", "project_id"]},
+    "sdk_updates": {"required_fields": ["organization_id"]},
+    "stacktrace_link": {"required_fields": ["organization_id", "project_id"]},
+    "suggest_mobile_project": {"required_fields": ["organization_id"]},
+    "suspect_commits": {"required_fields": ["organization_id", "project_id"]},
```
Comment (Zylphrex) [inline src/sentry/utils/prompts.py:14]: let's split this into a separate PR then merge + deploy it first

File: static/app/views/profiling/index.tsx [hunk 7]
```diff
@@ -1,23 +1,96 @@
...
 
 function ProfilingContainer({organization, children}: Props) {
-  function renderNoAccess() {
-    return (
-      <PageContent>
-        <Alert type="warning">{t("You don't have access to this feature")}</Alert>
-      </PageContent>
-    );
-  }
+  const client = useApi();
```
Comment (Zylphrex) [inline static/app/views/profiling/index.tsx:39]: nit: this is usually called `api` but no strong opinions either way

```suggestion
  const api = useApi();
```

**Review by Zylphrex — CHANGES_REQUESTED** [review]
let's make sure to add the prompts to the backend in a separate PR

File: static/app/views/profiling/index.tsx (hunk 4 above)
Comment (eliashussary) [reply static/app/views/profiling/index.tsx:70]: can we align `status` and `action` ? right now `action` is 'done | dismiss' but status is 'done | dismiss*ed*'?

File: static/app/utils/analytics/profilingAnalyticsEvents.tsx (hunk 5 above)
Comment (JonasBa) [reply static/app/utils/analytics/profilingAnalyticsEvents.tsx:10]: less code, works for me :D

**Review by Zylphrex — APPROVED** [review]
Just the minor comment around aligning `status` and `action` but otherwise LGTM
